src/main/java/Duke.java

```diff
@@ -1,10 +1,96 @@
+import java.util.Scanner;
+
 public class Duke {
+    
+    /**
+     * Prints a list of tasks with their status icons and descriptions.
+     *
+     * @param storageArray An array of Task objects representing the list of tasks.
+     * @param num          The number of tasks to be printed.
+     */
+
+    public static void printList(Task[] storageArray, int num) {
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < num; i++) {
+            System.out.println(
+                    i + 1 + "." + "[" + storageArray[i].getStatusIcon() + "]" + storageArray[i].getDescription());
+        }
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+        String logoRyan = " ____  _          \n"
+                + "|  _ \\| | ___   __\n"
+                + "| |_) | |/ _ \\ / _|\n"
+                + "|  __/| | (_) | (__\n"
+                + "|_|   |_|\\___/ \\___|\n";
+
+        System.out.println("Hello from\n" + logoRyan);
+
+        Task[] storageArray = new Task[100];
+        int numOfItems = 0;
+
+        String line = "Hello! I'm Ryan Loh \nWhat can I do for you?\n";
+
+        Scanner scanner = new Scanner(System.in);
+
+        String userInput = scanner.nextLine();
+
+        while (!userInput.equals("Bye")) {
+            String[] words = userInput.split(" ");
+
+            String initialWord = words[0];
+
+            if (initialWord.equals("list")) {
+
+                printList(storageArray, numOfItems);
+
+            } else if (initialWord.equals("mark")) {
+                System.out.println("mark " + words[1]);
+                int index = Integer.parseInt(words[1]);
+                // we need to check if its out of bounds first
+                if (index > numOfItems || index <= 0) {
```

Comment:
Consider separating the guard clause so that the main implementation is clearer
```suggestion
                if (index > numOfItems || index <= 0) {
                        continue;
                }
                // Mark the selected task
```